ct/bookstack.sh

```diff
@@ -60,30 +60,45 @@ check_container_resources
 if [[ ! -d /opt/bookstack ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
 RELEASE=$(curl -s https://api.github.com/repos/BookStackApp/BookStack/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
 if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+
   msg_info "Stopping Apache2"
-  systemctl stop apache2
+  systemctl stop apache2t 
   msg_ok "Services Stopped"
 
   msg_info "Updating ${APP} to ${RELEASE}"
-  cp /opt/bookstack/.env /opt/.env
+  cp -r /opt/bookstack/ /opt/bookstack-backup 
+  tar -czf /opt/bookstack_backup_${RELEASE}.tar.gz /opt/bookstack/.env /opt/bookstack/public/uploads /opt/bookstack/storage/uploads /opt/bookstack/themes &>/dev/null
+  mysqldump -u root bookstack > /opt/bookstack_backup_${RELEASE}.sql
+  rm -rf /opt/bookstack/*
   wget -q "https://github.com/BookStackApp/BookStack/archive/refs/tags/v${RELEASE}.zip"
   unzip -q v${RELEASE}.zip
-  mv BookStack-${RELEASE} /opt/bookstack
-  mv /opt/.env /opt/bookstack/.env
+  mv BookStack-${RELEASE}/* /opt/bookstack
+  cp /opt/bookstack-backup/.env /opt/bookstack/.env
+  cp -r /opt/bookstack-backup/public/uploads/ /opt/bookstack/public/uploads
+  cp -r /opt/bookstack-backup/storage/uploads/ /opt/bookstack/storage/uploads
+  cp -r /opt/bookstack-backup/themes/ /opt/bookstack/themes
   cd /opt/bookstack
   COMPOSER_ALLOW_SUPERUSER=1 composer install --no-dev  &>/dev/null
   php artisan key:generate --force &>/dev/null
   php artisan migrate --force &>/dev/null
+  chown www-data:www-data -R /opt/bookstack /opt/bookstack/bootstrap/cache /opt/bookstack/public/uploads /opt/bookstack/storage 
+  chmod -R 755 /opt/bookstack /opt/bookstack/bootstrap/cache /opt/bookstack/public/uploads /opt/bookstack/storage 
+  chmod -R 775 /opt/bookstack/storage /opt/bookstack/bootstrap/cache /opt/bookstack/public/uploads
+  chmod -R 640 /opt/bookstack/.env 
   echo "${RELEASE}" >/opt/${APP}_version.txt
   msg_ok "Updated ${APP}"
 
-  msg_info "Starting Apache2"
+  msg_info "Starting Apache2 "
   systemctl start apache2
   msg_ok "Started Apache2"
 
   msg_info "Cleaning Up"
-  rm -rf v${RELEASE}.zip
-  msg_ok "Cleaned"
+  cd /root/
```

Comment:
not needed?

Reply:
It is needed, we swicht to `cd /opt/bookstack` above. 
But i can remove it and change 
```
rm -rf /opt/bookstack-backup
rm -rf ${RELEASE}.zip
rm -rf BookStack-${RELEASE}
```
to
```
rm -rf /opt/bookstack-backup
rm -rf /root/${RELEASE}.zip
rm -rf /root/BookStack-${RELEASE}
```
But i find the above more descriptive.

Reply:
Or rm -rf ~/Filename ?

Reply:
Yeah, i like that. Thank you!